examples/stm32/xively_demo_stm32f4xx_nucleo_eth/Src/main.c

```diff
@@ -0,0 +1,367 @@
+  * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+  * PARTICULAR PURPOSE AND NON-INFRINGEMENT OF THIRD PARTY INTELLECTUAL PROPERTY
+  * RIGHTS ARE DISCLAIMED TO THE FULLEST EXTENT PERMITTED BY LAW. IN NO EVENT
+  * SHALL STMICROELECTRONICS OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+  * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+  * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+  * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+  * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+  * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+  * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+  *
+  ******************************************************************************
+  */
+/* Includes ------------------------------------------------------------------*/
+#include "main.h"
+#include "lwip/netif.h"
+#include "lwip/tcpip.h"
+#include "cmsis_os.h"
+#include "ethernetif.h"
+#include "app_ethernet.h"
+#include "httpserver-netconn.h"
+#include <xively_client.h>
+#include "demo_io.h"
+
+/* Private typedef -----------------------------------------------------------*/
+/* Private define ------------------------------------------------------------*/
+/* Private macro -------------------------------------------------------------*/
+/* Private variables ---------------------------------------------------------*/
+struct netif gnetif; /* network interface structure */
+
+/* UART handler declaration */
+UART_HandleTypeDef UartHandle;
+
+/* Private function prototypes -----------------------------------------------*/
+#ifdef __GNUC__
+/* With GCC/RAISONANCE, small printf (option LD Linker->Libraries->Small printf
+   set to 'Yes') calls __io_putchar() */
+#define PUTCHAR_PROTOTYPE int __io_putchar( int ch )
+#else
+#define PUTCHAR_PROTOTYPE int fputc( int ch, FILE* f )
+#endif /* __GNUC__ */
+
+/* Private function prototypes -----------------------------------------------*/
+static void SystemClock_Config( void );
+static void StartThread( void const* argument );
+static void Netif_Config( void );
+static void BSP_Config( void );
+
+/* Private functions ---------------------------------------------------------*/
+
+static void Error_Handler( void )
+{
+    /* Turn LED3 on */
+    BSP_LED_On( LED3 );
+    while ( 1 )
+    {
+    }
+}
+
+/**
+  * @brief  Main program
+  * @param  None
+  * @retval None
+  */
+int main( void )
+{
+    /* STM32F4xx HAL library initialization:
+         - Configure the Flash ART accelerator on ITCM interface
+         - Configure the Systick to generate an interrupt each 1 msec
+         - Set NVIC Group Priority to 4
+         - Global MSP (MCU Support Package) initialization
+       */
+    HAL_Init();
+
+    /* Configure the system clock to 180 MHz */
+    SystemClock_Config();
+
+    BSP_LED_Init( LED3 );

```

Comment:
Sorry but how things seem now this won't be priority. But I would prefer the first because it will be enough for the next 1 year. But if wind turns we can implement generic Xively Demo GPIO handling mechanism.